pillarbox-cast/src/main/java/ch/srgssr/pillarbox/cast/PillarboxCastPlayer.kt

```diff
@@ -109,6 +120,52 @@ class PillarboxCastPlayer(
         return castPlayer.isCastSessionAvailable
     }
 
+    override fun getTrackSelectionParameters(): TrackSelectionParameters {
+        return trackSelectionParameters
+    }
+
+    override fun getCurrentTracks(): Tracks {
+        return tracks
+    }
+
+    private fun getMediaStatus(): MediaStatus? {
+        return remoteMediaClient?.mediaStatus
+    }
+
+    override fun setTrackSelectionParameters(parameters: TrackSelectionParameters) {
+        if (remoteMediaClient == null && parameters == trackSelectionParameters) return
+        val oldParameters = this.trackSelectionParameters
+        this.trackSelectionParameters = parameters
+        notifyTrackSelectionParametersChanged()
+        val selectedTrackIds = trackSelector.getActiveMediaTracks(trackSelectionParameters, tracks = currentTracks)
+        remoteMediaClient?.setActiveMediaTracks(selectedTrackIds)?.setResultCallback {
+            if (!it.status.isSuccess) {
+                this.trackSelectionParameters = oldParameters
+                notifyTrackSelectionParametersChanged()
+            }
+        }
+    }
+
+    private fun updateCurrentTracksAndNotify() {
+        if (remoteMediaClient == null) return
+        val mediaTracks = getMediaStatus()?.mediaInfo?.mediaTracks ?: emptyList<MediaTrack>()
+        val tracks = if (mediaTracks.isEmpty()) {
+            Tracks.EMPTY
+        } else {
+            val selectedTrackIds: LongArray = getMediaStatus()?.activeTrackIds ?: longArrayOf()
+            val tabTrackGroup = Array<Tracks.Group>(mediaTracks.size) {
+                val mediaTrack = mediaTracks[it]
```

Comment:
Why use an intermediate array?
```suggestion
            val tabTrackGroup = mediaTracks.map { mediaTrack ->
```

Reply:
Because I copy-past from Media3 code :P